tomography_preprocessing/denoising/care4relion/care4relion.py

```diff
@@ -0,0 +1,149 @@
+    frames during Motion Correction (optimal) or by taking odd/even tilts during tomogram reconstruction.
+    The location of these tomograms should be specified in the global star file for all tilt series with the headers: 
+    
+     
+    _rlnTomoTomogramHalvesForDenoisingEven
+    
+      _rlnTomoTomogramHalvesForDenoisingOdd
+
+    Parameters
+    ----------
+    
+    tilt_series_star_file: RELION tilt-series STAR file.
+    
+    output_directory: directory in which results will be stored.
+    
+    training_tomograms: tomograms which are to be used for denoising neural network training.
+        User should enter tomogram names as rlnTomoName, separated by colons, e.g. TS_1:TS_2
+	
+    model_name (optional): if you have already trained a denoising neural network and want to skip the training step,
+        user can provide the path to the model.tar.gz produced by a previous cryoCARE denoising job. This will skip
+        straight to tomogram generation.
+	
+    n-tiles (optional): Initial tiles per dimension during prediction step. Should get increased \ 
+        if the tiles do not fit on the GPU. However, sometimes this parameter is a source of errors causing out of memory \
+        problems and tomogram dimensions to be wrong. We have found other useful values for this include 4,4,2 and 2,4,4 \
+        (default = 1,1,1). Enter as: `--n-tiles 4 4 2`
+        
+
+    Returns
+    -------
+    Denoised tomograms.
+    """
+    if not tilt_series_star_file.exists():
+        e = 'Could not find tilt series star file'
+        console.log(f'ERROR: {e}')
+        raise RuntimeError(e)    
+    
+    if model_name is None and training_tomograms is None:
+        e = 'Both model name and training tomograms are empty. Either provide a trained network through model name or specify \
+            which tomograms to train on (using rlnTomoName of the tilt series, separated by colons) in training tomograms'
+        console.log(f'ERROR: {e}')
+        raise RuntimeError(e)
+      
+    global_star = starfile.read(tilt_series_star_file, always_dict=True)['global']
+    
+    if not 'rlnTomoTomogramHalvesForDenoisingEven' in global_star.columns:
+        e = 'Could not find rlnTomoTomogramHalvesForDenoisingEven in tilt series star file.'
+        console.log(f'ERROR: {e}')
+        raise RuntimeError(e)
+    
+    training_dir, tomogram_dir, tilt_series_dir = \
+        create_denoised_tomograms_dir(output_directory)
+    
+    training_tomograms = parse_training_tomograms(training_tomograms)
+
+    training_tomograms_star = generate_training_tomograms_star(
+        global_star=global_star,
+        training_tomograms=training_tomograms,
+    )
+    
+    even_tomos, odd_tomos = find_tomogram_halves(training_tomograms_star)
+    
+    if model_name is None:
+        console.log('User has not provided path to a previously trained neural network, beginning to train now...')
```

Comment:
as above

Reply:
as above, will separate training and prediction